acs-integration-tests/src/test/java/com/ge/predix/integration/test/PolicyEvaluationCachingIT.java

```diff
@@ -0,0 +1,519 @@
+                        testPolicyName, this.acsZone1Headers);
+                this.privilegeHelper.deleteSubject(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(),
+                        subject.getSubjectIdentifier(), this.acsZone1Headers);
+            }
+        }
+    }
+
+    /**
+     * This test makes sure that cached policy evaluation results are properly invalidated when the name of a policy
+     * set changes as well as the policies it contains.
+     */
+    @Test
+    public void testPolicyEvalCacheWhenPolicySetNameChanges() throws Exception {
+        String testPolicyName = null;
+        String endpoint = this.acsUrl;
+        try {
+            String policyFile = "src/test/resources/policies/indeterminate.json";
+            testPolicyName = this.policyHelper.setTestPolicy(this.acsAdminRestTemplate, this.acsZone1Headers, endpoint,
+                    policyFile);
+            PolicyEvaluationRequestV1 policyEvaluationRequest = this.policyHelper.createEvalRequest(NOT_MATCHING_ACTION,
+                    DEFAULT_SUBJECT_ID, DEFAULT_RESOURCE_IDENTIFIER, null);
+            ResponseEntity<PolicyEvaluationResult> postForEntity = this.acsAdminRestTemplate.postForEntity(
+                    endpoint + PolicyHelper.ACS_POLICY_EVAL_API_PATH,
+                    new HttpEntity<>(policyEvaluationRequest, this.acsZone1Headers), PolicyEvaluationResult.class);
+            Assert.assertEquals(postForEntity.getStatusCode(), HttpStatus.OK);
+            PolicyEvaluationResult responseBody = postForEntity.getBody();
+            Assert.assertEquals(responseBody.getEffect(), Effect.NOT_APPLICABLE);
+            this.policyHelper.deletePolicySet(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(), testPolicyName,
+                    this.acsZone1Headers);
+
+            policyFile = "src/test/resources/policies/permit-admin-only-using-condition.json";
+            testPolicyName = this.policyHelper.setTestPolicy(this.acsAdminRestTemplate, this.acsZone1Headers, endpoint,
+                    policyFile);
+
+            policyEvaluationRequest = this.policyHelper.createEvalRequest(NOT_MATCHING_ACTION, DEFAULT_SUBJECT_ID,
+                    DEFAULT_RESOURCE_IDENTIFIER, null);
+            postForEntity = this.acsAdminRestTemplate.postForEntity(endpoint + PolicyHelper.ACS_POLICY_EVAL_API_PATH,
+                    new HttpEntity<>(policyEvaluationRequest, this.acsZone1Headers), PolicyEvaluationResult.class);
+
+            Assert.assertEquals(postForEntity.getStatusCode(), HttpStatus.OK);
+            responseBody = postForEntity.getBody();
+            Assert.assertEquals(responseBody.getEffect(), Effect.DENY);
+        } finally {
+            if (testPolicyName != null) {
+                this.policyHelper.deletePolicySet(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(),
+                        testPolicyName, this.acsZone1Headers);
+            }
+        }
+    }
+
```

Comment:
What's the difference between testPolicyEvalCacheWhenPolicySetNameChanges() and testPolicyEvalCacheWhenPolicySetChanges().  Both test for changes in the policy set.  Do we want to test just the policy set name?